shopify/product-reviews-template/api/actions/sendReviewRequests.js

```diff
@@ -0,0 +1,47 @@
+import { SendReviewRequestsGlobalActionContext } from "gadget-server";
+import { v4 as uuidv4 } from "uuid";
+
+/**
+ * @param { SendReviewRequestsGlobalActionContext } context
+ */
+export async function run({ params, logger, api, connections }) {
+  let orders = await api.shopifyOrder.findMany({
+    first: 250,
+    filter: {
+      fulfillmentStatus: {
+        equals: "fulfilled",
+      },
+      requestReviewAfter: {
+        lessThanOrEqual: new Date(),
+      },
+    },
+    select: {
+      singleUseCode: true,
+      customer: {
+        email: true,
+      },
+    },
+  });
+
+  let allOrders = orders;
+
+  while (orders.hasNextPage) {
+    orders = await orders.nextPage();
+    allOrders = allOrders.concat(orders);
+  }
+
+  const options = {
+    queue: {
+      name: `send-wishlist-emails-${uuidv4()}`,
+      maxConcurrency: 50,
+    },
+    retries: 1,
+  };
+
+  if (allOrders.length)
+    await api.enqueue(
```

Comment:
Doing a quick code check of this file -> enqueueEmails -> sendEmail it seems like we might send an email every time this is run? So customers will get multiple emails?

We don't appear to be tracking "email sent and code not used"

Reply:
I just fixed this